UndertaleModLib/Models/UndertaleTextureGroupInfo.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UndertaleModLib.Models
+{
+    public class UndertaleTextureGroupInfo : UndertaleObject
+    {
+        public UndertaleString GroupName;
+        public List<UndertaleResourceById<UndertaleEmbeddedTexture>> TexturePages;
+        public List<UndertaleResourceById<UndertaleSprite>> Sprites;
+        public List<UndertaleResourceById<UndertaleSprite>> SpineSprites;
+        public List<UndertaleResourceById<UndertaleFont>> Fonts;
+        public List<UndertaleResourceById<UndertaleBackground>> Tilesets;
+
+        public UndertaleTextureGroupInfo()
+        {
+            TexturePages = new List<UndertaleResourceById<UndertaleEmbeddedTexture>>();
+            Sprites = new List<UndertaleResourceById<UndertaleSprite>>();
+            SpineSprites = new List<UndertaleResourceById<UndertaleSprite>>();
+            Fonts = new List<UndertaleResourceById<UndertaleFont>>();
+            Tilesets = new List<UndertaleResourceById<UndertaleBackground>>();
+        }
+
+        public void Serialize(UndertaleWriter writer)
+        {
+            writer.WriteUndertaleString(GroupName);
+            uint pointer1 = writer.Position; writer.Write(0);
+            uint pointer2 = writer.Position; writer.Write(0);
+            uint pointer3 = writer.Position; writer.Write(0);
+            uint pointer4 = writer.Position; writer.Write(0);
+            uint pointer5 = writer.Position; writer.Write(0);
+
+            SeekWritePointer(writer, pointer1);
```

Comment:
This is quite repetitive. How about implementing something like UndertaleSimpleList that handles UndertaleResourceById instead of normal objects? UndertaleSimpleResourcesList? Could probably be used in a couple more places as well, this is not the first time we see a sequence like that.